keras_cv/models/densenet.py

```diff
@@ -0,0 +1,306 @@
+    if blocks == [6, 12, 48, 32]:
+        return "densenet201"
+    return "densenet"
+
+
+def _apply_classifier(x, classes, classifier_activation):
+    x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
+    x = layers.Dense(classes, activation=classifier_activation, name="predictions")(x)
+    return x
+
+
+def _apply_pooling_layer(x, pooling):
+    print(x.shape)
+    if pooling == "avg":
+        return layers.GlobalAveragePooling2D(name="avg_pool")(x)
+    if pooling == "max":
+        return layers.GlobalMaxPooling2D(name="max_pool")(x)
+    return x
+
+
+def DenseNet(
+    blocks,
+    include_preprocessing,
+    include_top=True,
+    weights=None,
+    input_tensor=None,
+    input_shape=(None, None, 3),
+    pooling=None,
+    classes=1000,
+    classifier_activation="softmax",
+):
+    """Instantiates the DenseNet architecture.
+
+    Reference:
+    - [Densely Connected Convolutional Networks (CVPR 2017)](
+        https://arxiv.org/abs/1608.06993)
+
+    This function returns a Keras DenseNet model.
+
+    For transfer learning use cases, make sure to read the
+    [guide to transfer learning & fine-tuning](
+      https://keras.io/guides/transfer_learning/).
+
+    Args:
+      blocks: numbers of building blocks for the four dense layers.
+      include_preprocessing: whether or not to Rescale the inputs.
+        If set to True, inputs will be passed through a
+        `Rescaling(1/255.0)` layer.
+      include_top: whether to include the fully-connected
+        layer at the top of the network.
+      weights: one of `None` (random initialization), or a pretrained
+        checkpoint.
+      input_tensor: optional Keras tensor
+        (i.e. output of `layers.Input()`)
+        to use as image input for the model.
+      input_shape: optional shape tuple, defaults to (None, None, 3).
+      pooling: optional pooling mode for feature extraction
+        when `include_top` is `False`.
+        - `None` means that the output of the model will be
+            the 4D tensor output of the
+            last convolutional block.
+        - `avg` means that global average pooling
+            will be applied to the output of the
+            last convolutional block, and thus
+            the output of the model will be a 2D tensor.
+        - `max` means that global max pooling will
+            be applied.
+      classes: optional number of classes to classify images
+        into, only to be specified if `include_top` is True, and
+        if no `weights` argument is specified.
+      classifier_activation: A `str` or callable. The activation function to use
+        on the "top" layer. Ignored unless `include_top=True`. Set
+        `classifier_activation=None` to return the logits of the "top" layer.
+        When loading pretrained weights, `classifier_activation` can only
+        be `None` or `"softmax"`.
+
+    Returns:
+      A `keras.Model` instance.
+    """
+    if not (weights in {None} or tf.io.gfile.exists(weights)):
```

Comment:
in {None} seems to be a weird check, should it just be "weight is None"?

Reply:
the idea is that in the future we can add all of our pretrained checkpoints here.